src/components/Navigator/Navigator.tsx

```diff
@@ -0,0 +1,159 @@
+import Dialog from "@material-ui/core/Dialog";
+import Downshift from "downshift";
+import hotkeys from "hotkeys-js";
+import React from "react";
+import { useIntl } from "react-intl";
+import cmp from "semver-compare";
+
+import { APP_VERSION } from "@saleor/config";
+import useLocalStorage from "@saleor/hooks/useLocalStorage";
+import useNotifier from "@saleor/hooks/useNotifier";
+import {
+  getActions,
+  getCatalog,
+  getCustomers,
+  getViews,
+  hasActions,
+  hasCatalog,
+  hasCustomers,
+  hasViews
+} from "./modes/utils";
+import NavigatorInput from "./NavigatorInput";
+import NavigatorSection from "./NavigatorSection";
+import { QuickSearchAction } from "./types";
+import useQuickSearch from "./useQuickSearch";
+
+const navigatorHotkey = "ctrl+k, command+k";
+const navigatorNotificationStorageKey = "notifiedAboutNavigator";
+
+function getItemOffset(
+  actions: QuickSearchAction[],
+  cbs: Array<typeof getViews>
+): number {
+  return cbs.reduce((acc, cb) => cb(actions).length + acc, 0);
+}
+
+const Navigator: React.FC = () => {
+  const [visible, setVisible] = React.useState(false);
+  const input = React.useRef(null);
+  const [query, mode, change, actions] = useQuickSearch(visible, input);
+  const intl = useIntl();
+  const notify = useNotifier();
+  const [notifiedAboutNavigator, setNotifiedAboutNavigator] = useLocalStorage(
+    navigatorNotificationStorageKey,
+    false
+  );
+
+  React.useEffect(() => {
+    hotkeys(navigatorHotkey, event => {
+      event.preventDefault();
+      setVisible(!visible);
+    });
+
+    if (cmp(APP_VERSION, "2.1.0") !== 1 && !notifiedAboutNavigator) {
+      notify({
+        text: intl.formatMessage({
+          defaultMessage:
+            "Our new feature to help you with your daily task. Run Navigator using Ctrl+K shortcut. (Cmd+K for Mac users)",
```

Comment:
Little grammar fix:

```suggestion
            "Our new feature to help you with your daily tasks. Run Navigator using Ctrl+K shortcut (Cmd+K for Mac users).",
```

I would check if we could actually use "⌘" symbol instead of "Cmd".  Also not a big deal but keyboard shortcuts looks nice formatted as code: `⌘+K`, maybe we could do something like that in this message?